server.lua

```diff
@@ -1,22 +1,60 @@
 if ox.inventory then
     local ox_inventory = exports.ox_inventory
 
+    local function isMoneyEnough(money, price)
+        if money < price then
+            local missingMoney = price - money
+            TriggerClientEvent('ox_inventory:notify', source, {
+                type = 'error',
+                text = ('Not enough money! Missing %s$'):format(missingMoney)
+            })
+            return false
+        else
+            return true
+        end
+    end
+
     RegisterNetEvent('ox_fuel:pay', function(price)
         assert(type(price) == 'number', ('Price expected a number, received %s'):format(type(price)))
 
         local money = ox_inventory:GetItem(source, 'money', false, true)
 
-        if money < price then
-            return TriggerClientEvent('ox_inventory:notify', source, {
-                type = 'error',
-                text = ('Not enough money! Missing %s'):format(price)
-            })
-        end
+        if not isMoneyEnough(money, price) then return false end
 
         ox_inventory:RemoveItem(source, 'money', price)
         TriggerClientEvent('ox_inventory:notify', source, {
             type = 'success',
             text = ('Paid %s'):format(price)
         })
     end)
+
+    RegisterNetEvent('ox_fuel:fuelCan', function(hasCan, price)
+        local money = ox_inventory:GetItem(source, 'money', false, true)
+        if not isMoneyEnough(money, price) then return false end
+        
+        if hasCan then
+            ox_inventory:RemoveItem(source, 'WEAPON_PETROLCAN', 1)
```

Comment:
Why remove the item and add it back instead of just setting durability metadata?

Reply:
Because I'm tired of life, going to push some changes

Reply:
> Why remove the item and add it back instead of just setting durability metadata?

Oh yeah, now I can remember why, petrolcan durability is always returning 1 for me, any idea?